p2panda-net/src/config.rs

```diff
@@ -70,3 +68,18 @@ impl Default for Config {
         }
     }
 }
+
+/// Configuration parameters for gossip overlays.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct GossipConfig {
+    /// Maximum gossip message size in bytes.
+    pub max_message_size: usize,
+}
```

Comment:
The configuration surface shrinks a lot with this PR (iroh's API doesn't accept a config object anymore), I thought we can add fine-configuration of HyParView and PlumTree whenever it's required in the future and only make this parameter configurable for now